undertale/models/item/finetune_embedding.py

```diff
@@ -51,147 +83,116 @@
     parser.add_argument(
         "--start-epoch", type=int, default=0, help="starting epoch number"
     )
-    parser.add_argument("-b", "--batch-size", type=int, default=8, help="batch size")
 
     arguments = parser.parse_args()
 
     undertale_logging.setup_logging()
 
-    os.makedirs(arguments.output, exist_ok=True)
+    device = "cuda" if torch.cuda.is_available() else "cpu"
+    if torch.cuda.is_available():
+        torch.cuda.set_device(0)
+        torch.set_float32_matmul_precision("high")
```

Comment:
We should chat about this more - I'm not opposed to lowering precision to make use of the tensor cores on our H100s but I didn't for the initial pretraining runs. I have to admit I don't fully understand the tradeoff here, but we should probably do the same thing everywhere.

What are your thoughts?